python/src/main.cpp

```diff
@@ -131,34 +137,32 @@ PYBIND11_MODULE(kp, m)
       m, "Tensor", DOC(kp, Tensor))
       .def(
         "data",
-        [](kp::Tensor& self) {
+        [np](kp::Tensor& self) {
             // Non-owning container exposing the underlying pointer
-            switch (self.dataType()) {
-                case kp::Tensor::TensorDataTypes::eFloat:
-                    return py::array(
-                      self.size(), self.data<float>(), py::cast(&self));
-                case kp::Tensor::TensorDataTypes::eUnsignedInt:
-                    return py::array(
-                      self.size(), self.data<uint32_t>(), py::cast(&self));
-                case kp::Tensor::TensorDataTypes::eInt:
-                    return py::array(
-                      self.size(), self.data<int32_t>(), py::cast(&self));
-                case kp::Tensor::TensorDataTypes::eDouble:
-                    return py::array(
-                      self.size(), self.data<double>(), py::cast(&self));
-                case kp::Tensor::TensorDataTypes::eBool:
-                    return py::array(
-                      self.size(), self.data<bool>(), py::cast(&self));
-                default:
-                    throw std::runtime_error(
-                      "Kompute Python data type not supported");
-            }
+            auto frombuffer = np.attr("frombuffer");
+
+            auto dtype =
+              dynamic_cast<PyTypeContainer*>(&(*self.dataType()))->clazz_;
```

Comment:
Not sure I follow - what is class_ in this case (same Q as above)?

Reply:
clazz_ should be renamed to dtype.

we need the dtype to cast the array from byte to the original dtype

Reply:
Fixed an issue causing tests to fail, the base object was being overwritten by frombuffer. fixed the implementation.